doc/setup.rst

```diff
@@ -198,6 +198,55 @@ To help expose these tracepoints (and any additional one we might require in
 the future) as trace events, an external module is required and is provided
 under the name of "lisa" in $LISA_HOME/tools/kmodules/lisa
 
+Pre-requisites
+.................
+
+CFI
+~~~~~
+
+First of all, building the module outside the kernel will not work if the kernel
+has CFI (Control-Flow Integrity) enabled as it will not allow loading of outside
+modules. Android kernels come with CFI by default so it will have to be disabled
+in the config (``CONFIG_CFI_CLANG``).
+If you want to use the module with a CFI-enabled kernel, follow the steps under
+"Integrating the module in your kernel tree" below.
+
+Kernel symbols needed for pixel6_emeter on Android product kernels
+~~~~~
+
+In order to use the pixel6_emeter on a product kernel, some symbols forbidden
+by Google need to be re-enabled. Without this step, the module will not load
+on a product kernel.
+
+Google product kernels (as of 5.10 for Pixel 6 & Pixel 7) are split into two
+source trees - ``aosp/`` and ``private/gs-google``. Changes to the base kernel
+code need to be made in the former, changes to vendor modules in the latter.
+If the ``aosp/`` tree is modified, the kernel needs to be rebuilt with:
+
+.. code-block:: sh
+    BUILD_AOSP_KERNEL=1 ./build_slider.sh
+
+If any of the symbols are changed, the kernel needs to be rebuilt with:
+
+.. code-block:: sh
+    ./update_symbol_list.sh
+
+* Firstly, apply the patch found in the Lisa source tree under
+  ``tools/kmodules/lisa-in-tree/0001-fs-Move-filp_open-kernel_read-write-outside-the-name.patch``
```

Comment:
Don't include the file name in that path, as this will likely change over time. Just say to apply all the patches found in ``tools/kmodules/lisa-in-tree``